src/ArmTemplates/Common/API/Clients/Abstractions/ApiClientBase.cs

```diff
@@ -0,0 +1,52 @@
+﻿// --------------------------------------------------------------------------
+//  <copyright file="ApiClientBase.cs" company="Microsoft">
+//      Copyright (c) Microsoft Corporation. All rights reserved.
+//  </copyright>
+// --------------------------------------------------------------------------
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Abstractions
+{
+    public abstract class ApiClientBase
+    {
+        public static string BaseUrl = "https://management.azure.com";
```

Comment:
You extracted the functionality from the EntityExtractorBase, but you also duplicated this static field that is now not set anywhere anymore. So while you overwrite the default EntityExtractorBase.BaseUrl, you always use the default url in this file.

Reply:
(related to the comment in EntityExtractorBase)
But again, can this not be a static field that's overwritten from the outside? It leads to issues exactly like this.